cmd/omegaup-grader/ci.go

```diff
@@ -97,49 +128,104 @@ func (h *ciHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	// Do the barest minimum checks before fully committing to making this CI
 	// run.
-	repository, err := git.OpenRepository(grader.GetRepositoryPath(
-		ctx.Config.Grader.RuntimePath,
-		report.Problem,
-	))
-	if err != nil {
-		ctx.Log.Error(
-			"failed to open repository",
-			map[string]interface{}{
-				"filename": reportPath,
-				"err":      err,
-			},
-		)
-		w.WriteHeader(http.StatusNotFound)
-		return
-	}
-	defer repository.Free()
-	commitID, err := git.NewOid(report.CommitHash)
-	if err != nil {
-		ctx.Log.Error(
-			"failed to parse commit",
-			map[string]interface{}{
-				"filename": reportPath,
-				"commit":   report.CommitHash,
-				"err":      err,
-			},
-		)
-		w.WriteHeader(http.StatusNotFound)
-		return
-	}
-	commit, err := repository.LookupCommit(commitID)
-	if err != nil {
-		ctx.Log.Error(
-			"failed to lookup commit",
-			map[string]interface{}{
-				"filename": reportPath,
-				"commit":   report.CommitHash,
-				"err":      err,
-			},
-		)
-		w.WriteHeader(http.StatusNotFound)
-		return
+
+	// todo(frcepeda): check with frontend for rate limiting
+
+	var problemFiles *common.ProblemFiles
+
+	if report.IsEphemeral {
+		r.ParseMultipartForm((base.Byte(150) * base.Mebibyte).Bytes())
```

Comment:
`todo` poner estos en un lugar más mono, pero no sé donde 🙃

Reply:
esto va a timeoutear bien recio :/ tenemos problemas para subir zips grandes, así que esto va a estar súper flaky.

además que andar mandando zips de más de 100 MiB nos va a salir bastante caro ^^;; así que realísticamente sólo podríamos permitir como uno de estos por minuto, lo cual suena medio inútil para fines de CI.